src/lib/components/Timeline.svelte

```diff
@@ -0,0 +1,64 @@
+<script lang="ts">
+  import { VersionControl } from '../versionControl';
+  import { onMount } from 'svelte';
+  // import { onMount, afterUpdate } from 'svelte';
+
+  const vcs = new VersionControl('src/lib/crc-decks/library.json');
+  let commits = [];
+
+  onMount(async () => {
+    // Simulate some commits for demonstration
+    await vcs.commit('Initial commit');
+    await vcs.commit('Update Book responsibilities');
+    await vcs.commit('Add Member collaborator to Report Class');
+    await vcs.commit('Add Book to Catalog collaborators');
+    commits = vcs.getCommits();
+  });
+
+  let hoveredCommit = null;
+</script>
+
+<div class="timeline relative pl-10 flex flex-col items-start h-full overflow-y-auto">
+  <div class="timeline-line"></div>
+  {#each commits as commit, index}
+    <div
+      class="timeline-item relative flex items-center mb-5 w-full z-10"
+      on:mouseenter={() => (hoveredCommit = commit)}
+      on:mouseleave={() => (hoveredCommit = null)}
+    >
+      <div class="timeline-point w-4 h-4 rounded-full bg-primary absolute -left-[12px]"></div>
+      <div class="timeline-content bg-base-300 rounded p-2 flex-grow">
+        <span class="commit-id font-bold text-xs block text-base-content">{commit.id}</span>
+        {#if hoveredCommit === commit}
+          <div class="commit-message mt-1 italic text-xs text-base-content">{commit.message}</div>
+        {/if}
```

Comment:
Ideally, the commit messages would always be visible. (Or maybe this is the idea of a /description/ from git? Where the message is the short-form and description is the long form?)

Reply:
interesting! got it. I do like the the idea of the commit messages being visible.